src/parser/GraphPattern.h

```diff
@@ -15,6 +15,12 @@ namespace parsedQuery {
 // Forward declarations.
 struct GraphPatternOperation;
 
+struct TextLimitMetaObject {
+  vector<Variable> _entityVars;
+  vector<Variable> _scoreVars;
```

Comment:
1. No always. Because of the fixed entity case the _scoreVars can be also longer.
2. Maybe it is best if we look at this struct together and find a suitable solution.